src/main/java/org/threadly/litesockets/TCPClient.java

```diff
@@ -36,7 +36,7 @@
   private static final ListenableFuture<Long> closedFuture = FutureUtils.immediateFailureFuture(new IllegalStateException("Connection is Closed"));
 
   private final MergedByteBuffers writeBuffers = new MergedByteBuffers();
-  private final Deque<Pair<Long, SettableListenableFuture<Long>>> writeFutures = new ArrayDeque<Pair<Long, SettableListenableFuture<Long>>>();
+  private final Deque<Pair<Long, SettableListenableFuture<Long>>> writeFutures = new ArrayDeque<Pair<Long, SettableListenableFuture<Long>>>(8);
```

Comment:
Actually it defaults to 16.  http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/util/ArrayDeque.java/#188